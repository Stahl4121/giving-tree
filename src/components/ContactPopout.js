import React, { useEffect } from 'react';

import Link from '@material-ui/core/Link';
import Popover from '@material-ui/core/Popover';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  a: {
    cursor: 'pointer',
  },
  popover: {
    maxWidth: 500,
  },
  typographyTop: {
    padding: theme.spacing(2),
  },
  typographyBottom: {
    paddingLeft: theme.spacing(2),
    paddingRight: theme.spacing(2),
    paddingBottom: theme.spacing(2),
  }
}));

export default function ContactPopout(props) {
  const classes = useStyles();
  const [anchorEl, setAnchorEl] = React.useState(null);

  const handleClick = event => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  const open = Boolean(anchorEl);
  const id = open ? 'Contact information' : undefined;

  return (
    <div>
      <a id="a-id" className={classes.a} aria-describedby={id} variant="contained" color="primary" onClick={handleClick}>
        <Button variant='outlined' color='primary' id="contact-button">
          Contact
        </Button>
      </a>
      <Popover
        className={classes.popover}
        id={id}
        open={open}
        anchorEl={anchorEl}
        onClose={handleClose}
        anchorOrigin={{
          vertical: 'top',
          horizontal: 'left',
        }}
        transformOrigin={{
          vertical: 'bottom',
          horizontal: 'left',
        }}
      >
        <Typography
          className={classes.typographyTop}
          variant="body2">
          Please contact <Link
            textDecoration='none'
            target='_blank'
            rel='noopener noreferrer'
            href={props.cfData.url}>
            {props.cfData.name}
          </Link> with any concerns at <Link href={'tel:' + props.cfData.phone}>
            {props.cfData.public_phone}
          </Link> or <Link href={'mailto:' + props.cfData.public_email}>
            {props.cfData.email}
          </Link>
        </Typography>
        <Typography
          className={classes.typographyBottom}
          variant="body2">
          <Link
            textDecoration='none'
            target='_blank'
            rel='noopener noreferrer'
            href={props.nonprofitData.url}>{props.nonprofitData.name}
          </Link> can be reached at <Link href={'tel:' + props.nonprofitData.phone}>
            {props.nonprofitData.phone}
          </Link> or <Link href={'mailto:' + props.nonprofitData.email}>
            {props.nonprofitData.email}
          </Link>
        </Typography>
      </Popover>
    </div>
  );
}