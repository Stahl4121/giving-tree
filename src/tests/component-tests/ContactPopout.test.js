import React from "react";
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ContactPopout from "../../components/ContactPopout.js";

describe('Component: ContactPopout', () => {
  var enzyme = require('enzyme');
  enzyme.configure({ adapter: new Adapter() });

  it('Renders correctly', () => {
    const mockCfData = "mockCfData";
    const mockNonprofitData = "mockNonprofitData";
    const wrap = enzyme.mount(<ContactPopout cfData={mockCfData} nonprofitData={mockNonprofitData} />)
    expect(wrap).toMatchSnapshot();
  });  

  it('contact button registers clicks', () => {
    const mockCfData = "mockCfData";
    const mockNonprofitData = "mockNonprofitData";
    const click = jest.fn();
    const wrap = enzyme.mount(<ContactPopout cfData={mockCfData} nonprofitData={mockNonprofitData} />)
    const contactButton = wrap.find('#contact-button').at(0);
    contactButton.props().onChange({ target: { value: "posting"} });
    expect(contactButton.mock.calls.length).toBe(1);
    expect(contactButton.mock.calls[0][0]).toBe("posting");
  });

})