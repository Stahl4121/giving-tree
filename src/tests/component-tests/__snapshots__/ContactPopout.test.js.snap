// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ContactPopout Renders correctly 1`] = `ReactWrapper {}`;
